hw/ip/i2c/doc/_index.md

```diff
@@ -243,10 +244,12 @@ Other features may also be required for complete SMBus functionality.)
 ### Clock Stretching
 As described in the I2C specification, a target device can pause a transaction by holding SCL low.
 A target device stretches the clock automatically when a host device reads from the target and the target's TX FIFO is empty.
-The present implementation also supports a mode in which a target device can stretch the clock after completing a transaction, if it needs more time, e.g., to store the acquired data byte in flash memory.
-If {{< regref STRETCH_CTRL.EN >}} is asserted, a target stretches the clock after an ingoing transaction completes.
-However, the target stretches the clock only if it is certain that the next transaction is intended for it and not for another target device on the bus.
-The target device stops stretching the clock and resumes normal operation when {{< regref STRETCH_CTRL.RESUME >}} is asserted.
+Also, a target device stretches the clock automatically when a host device writes to the target and the target's ACQ FIFO is full.
+The present implementation also supports few additional modes in which a target device can stretch the clock after completing a transaction, if it needs more time, e.g., to store the acquired data byte in flash memory, etc.
+If {{< regref STRETCH_CTRL.ENABLEADDR >}} is asserted, a target stretches the clock after acquiring an address and making certain that the transaction is intended for it and not for another device on the bus.
+If {{< regref STRETCH_CTRL.ENABLETX >}} is asserted, a target stretches the clock after an ongoing byte transmit operation (host is reading) completes and a host acknowledges the receipt.
```

Comment:
Though it doesn't need to be in this PR, these could *really* use some waveforms.  I had to re-read this at least four times to before I remembered what this was supposed to be.

Reply:
Added the waveforms.